File: app/src/main/java/com/popalay/cardme/ui/settings/SettingsActivity.java
package com.popalay.cardme.ui.settings;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;

import com.popalay.cardme.R;
import com.popalay.cardme.databinding.ActivitySettingsBinding;
import com.popalay.cardme.ui.base.BaseActivity;

public class SettingsActivity extends BaseActivity {

    private ActivitySettingsBinding b;

    public static Intent getIntent(Context context) {
        return new Intent(context, SettingsActivity.class);
    }

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        b = DataBindingUtil.setContentView(this, R.layout.activity_settings);
        initUI();
    }

    private void initUI() {
        setSupportActionBar(b.toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        b.toolbar.setNavigationOnClickListener(v -> finish());
    }
}

File: app/src/main/java/com/popalay/cardme/presentation/screens/settings/SettingsActivity.kt
package com.popalay.cardme.presentation.screens.settings

import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.content.Context
import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import com.popalay.cardme.R
import com.popalay.cardme.databinding.ActivitySettingsBinding
import com.popalay.cardme.presentation.base.BaseActivity
import dagger.android.AndroidInjection
import javax.inject.Inject

class SettingsActivity : BaseActivity() {

    @Inject lateinit var factory: ViewModelProvider.Factory

    private lateinit var b: ActivitySettingsBinding

    companion object {

        fun getIntent(context: Context) = Intent(context, SettingsActivity::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        b = DataBindingUtil.setContentView(this, R.layout.activity_settings)
        ViewModelProviders.of(this, factory).get(SettingsViewModel::class.java).let {
            b.vm = it
        }
        initUI()
    }


    private fun initUI() {
        setSupportActionBar(b.toolbar)
    }

}
